goinp/goinp_test.go

```diff
@@ -425,3 +428,227 @@ func TestSelectFromStringsFromReader(t *testing.T) {
+		t.Fatalf("invalid stdout: %s", stdout)
+	}
+
+	// char instead of number
+	answer, stdout, err = testAskOptionWrapper("title", "", false, "a\n2\n", "option one", "option two")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if answer != "option two" {
+		t.Fatalf("invalid value returned: %s", answer)
+	}
+	if stdout != `Select "title" from the list:
+[1] : option one
+[2] : option two
+Type in the option's number, then hit Enter: [31;1mfailed to parse option number, pick a number from 1-2[0m
+Type in the option's number, then hit Enter: ` {
+		t.Fatalf("invalid stdout: %s", stdout)
+	}
+
+	// empty instead of number
+	answer, stdout, err = testAskOptionWrapper("title", "", false, "\n2\n", "option one", "option two")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if answer != "option two" {
+		t.Fatalf("invalid value returned: %s", answer)
+	}
+	if stdout != `Select "title" from the list:
+[1] : option one
+[2] : option two
+Type in the option's number, then hit Enter: [31;1mfailed to parse option number, pick a number from 1-2[0m
+Type in the option's number, then hit Enter: ` {
+		t.Fatalf("invalid stdout: %s", stdout)
+	}
+
+	// auto select only element if required option
+	answer, stdout, err = testAskOptionWrapper("title", "", false, "the input can be whatever", "only one option auto selected")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if answer != "only one option auto selected" {
+		t.Fatalf("invalid value returned: %s", answer)
+	}
+	if stdout != `` {
+		t.Fatalf("invalid stdout: %s", stdout)
+	}
+
+	//// optional selector tests
+
+	// DO NOT auto select only element if optional option
+	answer, stdout, err = testAskOptionWrapper("title", "", true, "1\n", "only one option selected")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if answer != "only one option selected" {
+		t.Fatalf("invalid value returned: %s", answer)
+	}
+	if stdout != `Select "title" from the list:
+[1] : only one option selected
+[2] : <custom value>
+Type in the option's number, then hit Enter: ` {
+		t.Fatalf("invalid stdout: %s", stdout)
+	}
+
+	// optional option more element
+	answer, stdout, err = testAskOptionWrapper("title", "", true, "2\n", "option one", "option two")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if answer != "option two" {
+		t.Fatalf("invalid value returned: %s", answer)
+	}
+	if stdout != `Select "title" from the list:
+[1] : option one
+[2] : option two
+[3] : <custom value>
+Type in the option's number, then hit Enter: ` {
+		t.Fatalf("invalid stdout: %s", stdout)
+	}
+
+	// optional option more element and selected custom value
+	answer, stdout, err = testAskOptionWrapper("title", "", true, "3\nthis is my custom value\n", "option one", "option two")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if answer != "this is my custom value" {
+		t.Fatalf("invalid value returned: %s", answer)
+	}
+	if stdout != `Select "title" from the list:
+[1] : option one
+[2] : option two
+[3] : <custom value>
+Type in the option's number, then hit Enter: Enter value for "title": ` {
```

Comment:
NOTE: There is no new line between the two questions.

Reply:
there is when you are entering it in a terminal session

Reply:
Ok, thanks.